src/lib/functions/registry.js

```diff
@@ -0,0 +1,185 @@
+const chalk = require('chalk')
+
+const { NETLIFYDEVLOG, NETLIFYDEVERR } = require('../../utils/logo')
+const { mkdirRecursiveAsync } = require('../fs')
+const { getLogMessage } = require('../log')
+
+const { NetlifyFunction } = require('./netlify-function')
+const runtimes = require('./runtimes')
+const { watchDebounced } = require('./watcher')
+
+class FunctionsRegistry {
```

Comment:
This entity holds information about all registered functions. It's responsible for watching/unwatching files when functions are built. The registry uses a watcher for each function, which makes it a lot easier to know which functions to build when a given file changes.